sources/CMLPATHNAME

```diff
@@ -213,8 +209,13 @@ Copyright (c) 1986-1988, 1990 by Venue & Xerox Corporation.
 (DEFINEQ
 
 (PATHNAME
-(CL:LAMBDA (THING) (* hdj " 2-Apr-86 11:01") (* ;; "Turns Thing into a pathname.  Thing may be a string, symbol, stream, or pathname.") (CL:VALUES (CL:PARSE-NAMESTRING THING)))
-)
+  (CL:LAMBDA (THING)                                         (* ; "Edited 30-Apr-2023 13:59 by lmm")
+                                                             (* hdj " 2-Apr-86 11:01")
+
+         (* ;; "Turns Thing into a pathname.  Thing may be a string, symbol, stream, or pathname.")
+
+         (CL:UNLESS THING (CL:ERROR "~S not valid pathname" THING))
```

Comment:
I think perhaps this should be
```
(CL:CHECK-TYPE THING (AND (NOT NULL)
                          (OR STRING STREAM PATHNAME SYMBOL)))
```

SBCL says that the argument to PATHNAME should be `(OR STRING PATHNAME SYNONYM-STREAM FILE-STREAM)` but that doesn't seem quite right for us.
It's also not clear to me whether this check should be in `PATHNAME` or `PARSE-NAMESTRING`...